src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2225,4 +2231,50 @@ public static void makePhysNetNamesConsistent(Design design) {
         
         return path;
     }
+
+	/**
+	 * Create a Job running Vivado to create a readable version of
+	 * the EDIF inside the checkpoint to a separate file.
+	 * @param checkpoint the input checkpoint
+	 * @param edif the output EDIF
+	 * @return the created Job
+	 */
+	public static Job generateReadableEDIFJob(Path checkpoint, Path edif) {
+		try {
+
+			final Job job = new LocalJob();
+			job.setCommand(FileTools.getVivadoPath() + " -mode batch -source readable.tcl");
+
+			final Path runDir = Files.createTempDirectory(edif.toAbsolutePath().getParent(),edif.getFileName()+"_readable_edif_");
+			job.setRunDir(runDir.toString());
+
+			Files.write(runDir.resolve("readable.tcl"), Arrays.asList(
+					"open_checkpoint " + checkpoint.toAbsolutePath(),
+					"write_edif " + edif.toAbsolutePath()
+			));
+
+
+			return job;
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Use Vivado to create a readable version of the EDIF file inside a Checkpoint. Do nothing if the file already exists.
```

Comment:
The Javadoc specifies that it will do nothing if the EDIF file already exists, however I did not see any checking logic to support that statement.  In a quick test, it launched Vivado despite an EDIF file already being available.